### ISSUE TITLE
Reinforce searching with flags

### ISSUE THREAD

**Issue by sudo-nice** [vifm/vifm#232]
Please, consider these examples:

```
/pattern/X
?patter/X
```
where X could be one of:
```
h - number of hardlinks
s - size
i - number of items in a directory
a - atime
c - ctime
m - mtime
l - length of the file/directory name (with extension)
```

Had this implemented we could do powerful searches with minimal effort, like:
`?mp3$/s` - search for the smallest mp3-file;
`/img/n`  - search for the most populated directory, matching 'img' pattern;
`/./m`    - search for the most recent file (very handy for the directories where all the downloads store);

What do you think?

**Comment by xaizek** [comment]
There is definitely something to this idea, but I don't understand the exact mechanics of it and how would it combine with <kbd>n</kbd>/<kbd>N</kbd> keys.

In your examples flags pick either smallest or largest value, which is puzzling. It also seems that there can be only one result for each query (although there can be more than one file with largest size, for example).

Overall I get this as a combination of sorting and searching, but without changing order of files in the view. Is my understanding correct?

**Comment by sudo-nice** [comment]
Yes, it's kind of combination of searching and sorting. Let's look into details:

### Example: `?mp3$/s`

It should search for `mp3$` **from the very bottom** of the file-list. So, the current cursor position should not be respected.

1. If there is no mp3-file found, do nothing (the same like vifm acts now).
2. If there is one mp3-file, stay on it.
3. If there are several mp3-files, stay on the smallest one.
4. If <kbd>n</kbd> is pressed, jump to the second smallest mp3-file... and so on.
5. If <kbd>N</kbd> is pressed, move to the previous smallest mp3-file.

### Example: `/img/n`

It should serch for a directory (or a file, it doesn't matter) `img` **from the very top** of the file-list. The current cursor position is not respected again.

1. If there is no matched files/directories, do nothing.
2. If there is only one match , stay on it.
3. If there are several files/directories, matched pattern, jump to the match with most population.
4. <kbd>n</kbd> will jump to the second most populated directory.
5. <kbd>N</kbd> will jump to the previous rated directory.

Have I clarified the things?

**Comment by xaizek** [comment]
Thanks, that's more details. So `?` requests descending ordering. One more thing about <kbd>n</kbd>/<kbd>N</kbd> keys, which is ambiguous. If we have:

 * `01. title. mp3` of size 123
 * `02. title. mp3` of size 100
 * `03. title. mp3` of size 123

and do **/mp3$/s**, which selects first file (`01`).  Would <kbd>n</kbd> just cycle through `01` and `03` or go `01` -> `03` -> `02` and then backwards in reversed order?

**Comment by sudo-nice** [comment]
I think, `01` -> `03` -> `02` and then wrap again is more consistent acting, as we expect to iterate all over the matched files and unlikely want to miss some of them.

**Comment by xaizek** [comment]
OK, this makes it an extension of search that just changes order of file traversal. I think it's worth implementing, should be more convenient than changing sorting just to find a single file. Thanks for the idea.
